docs/source/guide/custom_metric.md

```diff
@@ -105,4 +107,137 @@ After adding your code to Label Studio Enterprise, the following could happen:
\ No newline at end of file
+If you change the labeling configuration for your project, you might need to update the custom agreement metric code to handle the new format of annotations produced.
+
+## Set up permissions for a private cloud custom agreement metric
+
+If you have Label Studio Enterprise deployed in a private cloud (self-managed) Amazon Web Services (AWS) Elastic Compute Cluster (EC2) instance or Amazon Elastic Kubernetes Service (EKS), you must grant additional permissions for the `label-studio` service account to be able to set up and run custom agreement metrics. 
+
+To set up the permissions, do the following: 
+1. Create an AWS IAM role to be used by the custom metric Lambda functions to store logs in Cloudwatch 
+2. Define an IAM policy allowing Label Studio Enterprise to run code in AWS Lambda.
+3. Attach the IAM policy to the `label-studio` service account. How you do this depends on your deployment scenario:
+   - [Deployed with Docker Compose running in EC2](#Deployed-with-Docker-Compose-running-in-EC2)
+   - [Deployed in EKS with an OIDC provider](#Deployed-in-EKS-with-an-OIDC-provider)
+   - [Deployed in EKS without an OIDC provider](#Deployed-in-EKS-without-an-OIDC-provider)
+
+You must know your AWS account ID to perform these steps. 
+
+### Create an AWS IAM role
+
+Using your preferred method, create an AWS IAM role. 
+
+1. Create an AWS IAM role named `LSE_CustomMetricsExecuteRole`. For guidance, see the [AWS Identity and Access Management doc for Creating IAM roles](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_create.html)
+2. Attach the following IAM policy to allow the role to store logs in Cloudwatch. Replace `YOUR_AWS_ACCOUNT` with your AWS account ID.
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "VisualEditor0",
+            "Effect": "Allow",
+            "Action": "logs:CreateLogGroup",
+            "Resource": "arn:aws:logs:*:YOUR_AWS_ACCOUNT:*"
+        },
+        {
+            "Sid": "VisualEditor1",
+            "Effect": "Allow",
+            "Action": [
+                "logs:CreateLogStream",
+                "logs:PutLogEvents"
+            ],
+            "Resource": [
+                "arn:aws:logs:*:YOUR_AWS_ACCOUNT:log-group:/aws/lambda/custom-metric-*"
+            ]
+        }
+    ]
+}
+```
+
+### Define an IAM policy
+
+After creating an IAM role to manage logs for the custom agreement metric, define an IAM policy to allow Label Studio Enterprise to interact with AWS Lambda. If you haven't created an IAM policy in AWS before, review the AWS Identity and Access Management documentation on [Creating IAM policies](https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies_create.html). 
+
+Create an IAM policy called `LSE_AllowInteractLambda` and replace `YOUR_AWS_ACCOUNT` with your AWS account ID:
```

Comment:
is it enough to just say "hey create this policy" without saying more about how or where? I punted to the AWS docs in that case.

Reply:
agreed, I've haven't provided detailed steps. Let me put these steps as a separate comment in this PR, and we will use it on this page and provide an anchor for multiple places where I refer to "create iam policy and attach"